app/airview_api/search.py

```
@@ -0,0 +1,106 @@
+    """Raised when an unknown backend is configured."""
+
+
+class SearchBackend:
+    _client = None
+
+    @property
+    def client(self):
+        if self._client is None:
+            self.create_client()
+        return self._client
+
+    def create_client(self):
+        raise NotImplementedError("create_client() must be implemented in child class.")
+
+    def query(self, search_term: str):
+        raise NotImplementedError("query() must be implemented in child class.")
+
+    def serialize(self, data):
+        raise NotImplementedError("serialize() must be implemented in child class.")
+
+
+class ElasticsearchBackend(SearchBackend):
+    def __init__(self, host: str, api_key: str, index: str, port: str = "9200"):
+        super(ElasticsearchBackend, self).__init__()
+        self.host = f"{host}:{port}"
+        self.api_key = api_key
+        self.index = index
+
+    def create_client(self):
+        from elasticsearch import Elasticsearch
+        self._client = Elasticsearch(
+            [self.host],
+            api_key=self.api_key,
+            use_ssl=SEARCH_BACKEND_ELASTIC_SSL_ENABLED,
+            verify_certs=SEARCH_BACKEND_ELASTIC_SSL_VERIFY_CERTS
+        )
+        return self._client
+
+    def serialize(self, data):
+        output = []
+        try:
+            actual_data = data['hits']['hits']
+            if isinstance(actual_data, list):
+                for payload in actual_data:
+                    output.append(payload['_source'])
+            elif isinstance(actual_data, dict):
+                output = actual_data['_source']
+        except KeyError:
+            return []
+        return output
+
+    def query(self, search_term: str):
+        body = {
+            "query": {
+                "multi_match": {  # This will returned a ranked search.
+                    "query": search_term
+                }
+            }
+        }
+
+        search_result = self.client.search(
```

Will need to handle exceptions (i.e. Timeouts)

---

Let it raise a 500 for now

---

500 is fine, not sure what else we can do about timeouts anyway and the frontend will put up a friendly 'unexpected error, try again' for it anyway, which is all the user can really do.